File: day2/Quiz1/solution.py
# maxDiff([1, 2, 3, 4]);    //return 3, because 4 - 1 == 3
# maxDiff([1, 2, 3, -4]);   //return 7, because 3 - (-4) == 7
# maxDiff([ 1, 2, 3, 4, 5, 5, 4]) // return 4
# maxDiff([-4, -5, -3, -1, -31])  // return 30
# maxDiff([1000000])  // return 0
# maxDiff([])         // return 0

def maxDiff(lst):
    
    return max(lst) - min(lst) if lst else 0

def maxDiff_old(lst):

    if len(lst) == 0:
        lst = [0]
    return max(lst) - min(lst)

if __name__ == "__main__":
    lst = []

    print(maxDiff(lst))

File: day2/Quiz1/solution.java
import java.util.Arrays;
import java.util.List;

public class solution {

    public static void main(String[] args) {

        //List<Integer> lst = Arrays.asList(1, 2, 3, -4);
        List<Integer> lst = List.of(1, 2, 3, -4); //java9~
        System.out.println(maxDiff(lst));

    }

    public static int maxDiff(List<Integer> lst) {
        return lst.stream().max(Integer::compare).orElse(0) - lst.stream().min(Integer::compare).orElse(0);
    }

    public static int maxDiff_old(List<Integer> lst) {
        int result = 0;
        
        if(lst.size() >= 2){
            int max = lst.get(0);
            int min = lst.get(0);

            for (int num : lst) {
                max = num > max ? num : max;
                min = num < min ? num : min;
            }
            result = max - min;
        }

        return result;
    }
} 